src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -210,15 +209,206 @@ module Make (Inputs : Inputs_intf.S) = struct
           acc.current <- f acc.current ;
           acc.next <- f acc.next )
 
-    let self_set_hash t address hash =
+    let self_set_hash_impl t address hash =
       update_maps t ~f:(fun maps ->
           { maps with hashes = Map.set maps.hashes ~key:address ~data:hash } )
 
+    let path_batch_impl ~fixup_path ~self_lookup ~base_lookup locations =
```

Comment:
Note to myself: slightly rewritten to not have `t` as an argument, and to not pass `hashes` to the `self_lookup` parameter (it's passed/embedded into `self_lookup` on the caller level).